internal/config/validator.go

```diff
@@ -44,14 +44,35 @@ func (v *validator) Validate(resources ...client.ObjectList) error {
 			clusterResources.Pools = append(clusterResources.Pools, list.Items...)
```

Comment:
I disagree because I don't understand why it is better than the existing one.

Reply:
I think it would be better to move the override logic to a function because this will make the switch-case block less cluttered so it is easier to read and understand.
We do extra logic for peers, bgpadv, and addresspool, and it might confuse the reader.
With a function the code will look cleaner, and we can add a small comment explaining why we do this non-trivial override.

Reply:
I agree that an additional function should be created whose role will only be to reset the fields that cause TransientError.